test/jdk/java/lang/Class/forName/NonJavaNames.java

```diff
@@ -44,9 +57,50 @@ public Baz2(){}
         return new Baz2();
      }
 
-    public static void main(String[] args) throws Exception {
-        NonJavaNames.Baz bz = new NonJavaNames.Baz();
+    private static final String SRC_DIR = System.getProperty("test.src");
+    private static final String TEST_SRC = SRC_DIR + "/classes/";
+    private static final String TEST_CLASSES = System.getProperty("test.classes", ".");
+    Path dhyphenPath, dcommaPath, dperiodPath, dleftsquarePath, drightsquarePath, dplusPath, dsemicolonPath, dzeroPath, dthreePath, dzadePath;
+
+    @BeforeClass
+    public void createInvalidNameClasses() throws IOException {
+        Path hyphenPath = Paths.get(TEST_SRC + "hyphen.class");
```

Comment:
Building a `Path` with string concatenation is an anti-pattern.
Also `Path.of` is probably preferred instead of `Paths.get`.

Reply:
Now not doing the concatenation of the strings to create Path.